### ISSUE TITLE
"exhaustive" linter breaks staticcheck

### ISSUE THREAD

**Issue by pierrre** [golangci/golangci-lint#1900]
- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).
- [ ] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

<details><summary>Description of the problem</summary>

When I enable the `exhaustive` linter, `staticcheck` is broken:
- it reports new errors that it didn't report previously
- it stops reporting some errors

</details>

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version 1.39.0 built from 9aea4aee on 2021-03-26T08:02:53Z
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
run:
  timeout: "10m"
output:
  sort-results: true
linters:
  disable-all: true
  enable:
    - "asciicheck"
    - "bodyclose"
    - "deadcode"
    - "depguard"
    - "errcheck"
    - "errorlint"
    - "exhaustive"
    - "exportloopref"
    # - "gocognit"
    - "gocritic"
    - "gocyclo"
    - "godot"
    - "goerr113"
    - "gofmt"
    - "gofumpt"
    - "goimports"
    - "golint"
    - "gosimple"
    - "govet"
    - "ineffassign"
    - "misspell"
    - "nakedret"
    # - "nestif"
    - "nilerr"
    - "nolintlint"
    - "staticcheck"
    - "structcheck"
    # - "stylecheck"
    - "unconvert"
    - "unparam"
    - "unused"
    - "varcheck"
linters-settings:
  depguard:
    list-type: blacklist
    include-go-root: true
    packages-with-error-message:
      - errors: "use github.com/xxx/yyy/errors"
      - io/ioutil: "use io or os"
      - reflect: "shouldn't be used by most application"
      - unsafe: "it's not safe"
      - github.com/pkg/errors: "use github.com/xxx/yyy/errors"
      - golang.org/x/net/context: "use context"
      - golang.org/x/sync/singleflight: "use github.com/xxx/yyy/singleflight"
  errcheck:
    check-type-assertions: true
  gocritic:
    enabled-tags:
      - diagnostic
      - style
      - performance
      - opinionated
      - experimental
    disabled-checks:
      - commentFormatting # TODO enable in order to follow the Go convention.
      - hugeParam # Too many problem for now. And maybe it's not a real issue.
      - paramTypeCombine # Too many false positive.
      - ptrToRefParam # TODO evaluate if this check can be enabled, and disable with nolint for specific cases.
      - rangeValCopy # TODO evaluate if this should be fixed or not.
      - tooManyResultsChecker # It's usually justified.
      - unnamedResult # TODO fix projects.
      - whyNoLint # It's the developer/reviewer responsibility.
  gocyclo:
    min-complexity: 10
  govet:
    enable-all: true
    disable:
      - fieldalignment
    settings:
      printf:
        funcs:
          - "(github.com/xxx/yyy/errors).Newf"
          - "(github.com/xxx/yyy/errors).WithMessagef"
          - "(github.com/xxx/yyyerrors).Wrapf"
  nolintlint:
    allow-unused: false
    allow-leading-space: false
    allow-no-explanation:
      - errcheck
      - misspell
    require-explanation: true
    require-specific: true
issues:
  exclude-use-default: false
  max-issues-per-linter: 0
  max-same-issues: 0

```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.16.3 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/pierre/.cache/go-build"
GOENV="/home/pierre/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/pierre/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/pierre/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/pierre/.gimme/versions/go1.16.3.src"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/pierre/.gimme/versions/go1.16.3.src/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.3"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/pierre/gosrc/github.com/xxx/yyy/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build4080227508=/tmp/go-build -gno-record-gcc-switches"
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

<details><summary>Code example or link to a public repository</summary>

```go
// add your code here
```

</details>

**Comment by bombsimon** [comment]
Why did you remove the issue template? The template has sections where you're supposed to provide code (or link to code) to reproduce this behavior. It also has a section where you're supposed to add the output from when you run `golangci-lint`, your version of Go, what configuration you're using etc. Without those it's really hard to troubleshoot and help you.

Could you please pull [the template](https://github.com/golangci/golangci-lint/issues/new?assignees=&labels=bug&template=bug_report.md) and edit your post to include this information?

**Comment by pierrre** [comment]
> Why did you remove the issue template?

Because some of the config contain info specific to the company I'm working for, and I don't want to show it.
I will try to redact it.

**Comment by pierrre** [comment]
Sorry, I did everything, and the issue seems fixed.
It was probably a cache issue.
Sorry for the false alert.
